### PR TITLE
Add testcase and change user name function

### PR DESCRIPTION
**Pull request by masayukipy**
Please review this @hiroyukikumazawa .

### CODE REVIEW

File: programs/game/src/lib.rs [hunk 1]
```diff
@@ -17,6 +17,15 @@ pub mod game {
         user_stats.bump = ctx.bumps.user_stats;
         Ok(())
     }
+    // handler function (add this next to the create_user_stats function in the game module)
```
Comment (hiroyukikumazawa) [inline programs/game/src/lib.rs]: update this comment

File: programs/game/src/lib.rs [hunk 2]
```diff
@@ -17,6 +17,15 @@ pub mod game {
...
+    pub fn change_user_name(ctx: Context<ChangeUserName>, new_name: String) -> Result<()> {
```
Comment (hiroyukikumazawa) [inline programs/game/src/lib.rs:20]: add comment

File: programs/game/src/lib.rs [hunk 3]
```diff
@@ -17,6 +17,15 @@ pub mod game {
...
+        if new_name.as_bytes().len() > 200 {
+            // proper error handling omitted for brevity
```
Comment (hiroyukikumazawa) [inline programs/game/src/lib.rs:22]: update this comment

File: tests/game.ts [hunk 4]
```diff
@@ -1,16 +1,46 @@
...
-  // Configure the client to use the local cluster.
-  anchor.setProvider(anchor.AnchorProvider.env());
+describe("game", async () => {
+    const provider = anchor.AnchorProvider.env();
+    anchor.setProvider(provider);
 
-  const program = anchor.workspace.Game as Program<Game>;
+    const program = anchor.workspace.Game as Program<Game>;
 
-  it("Is initialized!", async () => {
```
Comment (hiroyukikumazawa) [inline tests/game.ts:11]: split to two testcase

File: tests/game.ts [hunk 5]
```diff
@@ -1,16 +1,46 @@
...
-    // Add your test here.
-    const tx = await program.methods.initialize().rpc();
-    console.log("Your transaction signature", tx);
-  });
+    it("Sets and changes name!", async () => {
```
Comment (hiroyukikumazawa) [inline tests/game.ts:13]: low comment,
add full comment line by line

File: programs/game/src/lib.rs
Comment (hiroyukikumazawa) [inline programs/game/src/lib.rs:1]: totally great job

File: tests/game.ts
Comment (hiroyukikumazawa) [inline tests/game.ts:1]: ❤

**Review by hiroyukikumazawa — CHANGES_REQUESTED** [review]
Please fix these

File: programs/game/src/lib.rs (hunk 1 above)
Comment (masayukipy) [reply programs/game/src/lib.rs]: ok

File: programs/game/src/lib.rs (hunk 2 above)
Comment (masayukipy) [reply programs/game/src/lib.rs:20]: ok

File: tests/game.ts (hunk 4 above)
Comment (masayukipy) [reply tests/game.ts:11]: ok

File: tests/game.ts (hunk 5 above)
Comment (masayukipy) [reply tests/game.ts:13]: omg, ok

File: programs/game/src/lib.rs
Comment (masayukipy) [reply programs/game/src/lib.rs:1]: thanks

**Review by masayukipy — COMMENTED** [review]
cool

**Pull request merged**